### DISCUSSION TITLE
Type definitions for single file package

### DISCUSSION THREAD

**Discussion by remcohaszing** [python/typing#1297] [Q&A] [answered]
I would like to make [`pywakeonlan`](https://github.com/remcohaszing/pywakeonlan) work with typings. Currently this package ships a single file `wakeonlan.py`. It seems that I would need to change the file structure to the following:

```
.
├── LICENSE
├── pyproject.toml
├── README.rst
└── wakeonlan
    ├── __init__.py
    └── py.typed
```

I like that people can just download `wakeonlan.py` and run it, so ideally I would like to keep the `wakeonlan.py` file as-is in the project root. Is that possible?

**Comment by erictraut** [comment, marked as answer]
Your assumption is correct. You would need to switch to a directory structure to include a `py.typed` marker file. There is no provision in PEP 561 for marking a single-file package as typed.

**Reply by remcohaszing** [reply]
It's not the answer I was hoping for, but it is the anwer I was looking for. Thank you!

**Reply by not-my-profile** [reply]
For single-file `.py` packages using `__init__.py` works quite well. However the same cannot be said about packages that are implemented by a single shared library (`.so`). I just opened  #1333 about that. (commenting here so that people interested in this topic find that discussion :)
